lighthouse-core/audits/metrics/max-potential-fid.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit');
+const i18n = require('../../lib/i18n');
+
+const UIStrings = {
+  /** The name of the metric "Maximum Potential First Input Delay" that marks the maximum estimated time between the page receiving input (a user clicking, tapping, or typing) and the page responding. Shown to users as the label for the numeric metric value. Ideally fits within a ~40 character limit. */
+  title: 'Max Potential FID',
+  /** Description of the Estimated Input Latency metric that estimates the amount of time, in milliseconds, that the app takes to respond to user input. This description is displayed within a tooltip when the user hovers on the metric name to see more. No character length limits. 'Learn More' becomes link text to additional documentation. */
+  description: 'The potential First Input Delay that your users could experience is the ' +
+      'duration, in milliseconds, of the longest task.'
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class MaxPotentialFID extends Audit {
```

Comment:
can we get some overview comment to explain the basics of how this metric is calculated?

Reply:
done